rotkehlchen/accounting/cost_basis/base.py

```diff
@@ -428,6 +514,130 @@ def spend_asset(
         self.reduce_asset_amount(asset=asset, amount=amount, timestamp=timestamp)
         return None
 
+    def calculate_spend_average_cost_basis(
+            self,
+            spending_amount: FVal,
+            spending_asset: Asset,
+            timestamp: Timestamp,
+    ) -> CostBasisInfo:
+        """
+        Calculates the average cost base of the spend.
+
+        Formulae is taken from:
+        https://www.adjustedcostbase.ca/blog/how-to-calculate-adjusted-cost-base-acb-and-capital-gains/
+        """
+        remaining_sold_amount = spending_amount
+        taxfree_bought_cost = taxable_bought_cost = taxable_amount = taxfree_amount = ZERO  # noqa: E501
+        matched_acquisitions = []
+        asset_events = self.get_events(spending_asset)
+
+        acquisitions_manager: AverageCostBaseAcquisitionOrder = cast(AverageCostBaseAcquisitionOrder, asset_events.acquisitions_manager)  # noqa: E501
+
+        for acquisition_event in acquisitions_manager.processing_iterator():
+            # calculate the average cost base of the acquisition
+            # [Previous Total ACB] + [Cost of New Shares] + [Transaction Costs]
+            acquisitions_manager.current_cost_base += (acquisition_event.rate * acquisition_event.amount)  # noqa: E501
```

Comment:
> Did I explain my thoughts clearly enough?

Yes, you did. However, calculating the average cost basis for all the acquisitions before accounting for a spend sort of isn't how ACB works -- at least to my understanding.

When an acquisition happens
<img width="628" alt="image" src="https://user-images.githubusercontent.com/72208758/200369780-a331f028-89c5-497c-b309-a9b3ac22b435.png">


When a spend happens
<img width="624" alt="image" src="https://user-images.githubusercontent.com/72208758/200370037-3b93db13-4245-4d89-8011-7c847e76a67c.png">

Otherwise, what you said is easier in the sense that...
**get the running average cost base for all acquisitions, then deduct from this cost base whenever a spend happens.**
but then that'd be a wrong assumption.

Reply:
Yeah, I do that.

Reply:
Could you point to the place where this happens? Perhaps I then misread something

Reply:
https://github.com/prettyirrelevant/rotki/blob/30fdb098d1ffff7264ffd915039cb288bd46440c/rotkehlchen/accounting/cost_basis/base.py#L539

Reply:
Or wait... ohhh ohhh I think I get you now.

For every time `add_acquisition()` is called, update the cost base.
Then when `calculate_spend_average_cost_basis` is called, update the cost base for a sell.

meaning currently...

If I have two concurrent buys, it only uses the cost base of the first buy & does not take into account the second buy right? -- which it should

Reply:
Yeah! this what I meant